navigator_engine/common/graph_visualizer.py

```diff
@@ -0,0 +1,209 @@
+            graph = model.load_graph(graph_id=graph_id)
+            graph_x = graph.to_networkx()
+
+        root_nodes = [n for n, d in graph_x.in_degree() if d == 0]
+        assert len(root_nodes) == 1
+
+        elements = []
+        for n in graph_x.nodes:
+            node = model.load_node(node_id=n.id)
+            if node.action and node.action.complete:
+                node_element = {'data': {'id': str(node.id),
+                                         'label': f'A {node.action.id} COMPLETE',
+                                         'infobox': f'Action {node.action.id} | '
+                                                    f'Node {node.id} | '
+                                                    f'{node.action.title}'
+                                         },
+                                'classes': 'red triangle'
+                                }
+            elif node.action and not node.action.complete:
+                node_element = {'data': {'id': str(node.id),
+                                         'label': f'A {node.action.id}',
+                                         'infobox': f'Action {node.action.id} | '
+                                                    f'Node {node.id} | '
+                                                    f'{node.action.title}'
+                                         },
+                                'classes': 'red triangle'
+                                }
+            elif node.milestone:
+                node_element = {'data': {'id': str(node.id),
+                                         'label': f'M {node.milestone.id}',
+                                         'infobox': f'Milestone {node.milestone.id} | '
+                                                    f'Node {node.id}  | '
+                                                    f'{node.milestone.title}'
+                                         },
+                                'classes': 'blue square'
+                                }
+            elif node.conditional:
+                node_element = {'data': {'id': str(node.id),
+                                         'label': f'C {node.conditional.id}',
+                                         'infobox': f'Conditional '
+                                                    f'{node.conditional.id} | '
+                                                    f'Node {node.id}  | '
+                                                    f'{node.conditional.title}'
+                                         },
+                                'classes': 'green circle'
+                                }
+            else:
+                continue
+
+            elements.append(node_element)
+
+        for e in graph_x.edges:
+            edge = model.load_edge(from_id=e[0].id, to_id=e[1].id)
+            if edge.type:
+                edge_element = {'data': {'source': str(e[0].id), 'target': str(e[1].id)},
+                                'classes': 'true'}
```

Comment:
I'm just not a fan of this "hanging" multiline format.  I don't think it's very readable.  The general principle is that the closing parenthesis should always be at the same indent as the line the block was opened on. The extra line creating a bit more white space is only a good thing in my head.  Doing it this way also shortens the line lengths which is good as well. 
```suggestion
                edge_element = {
                    'data': {'source': str(e[0].id), 'target': str(e[1].id)},
                    'classes': 'true'
                }
```